apps/mymedicare_cb/models.py

```diff
@@ -57,43 +57,52 @@ def get_and_update_user(subject, mbi_hash, hicn_hash, first_name, last_name, ema
     # Match a patient identifier via the backend FHIR server
     fhir_id, hash_lookup_type = match_fhir_id(mbi_hash=mbi_hash, hicn_hash=hicn_hash, request=request)
 
+    # Init for types of crosswalk updates. "" = None
+    crosswalk_updated = ""
+
     try:
         # Does an existing user and crosswalk exist for SLSx username?
         user = User.objects.get(username=subject)
 
-        if user.crosswalk.user_hicn_hash != hicn_hash:
-            err_msg = "Found user's hicn did not match"
-            log_get_and_update_user(auth_flow_dict, "FAIL", user, fhir_id, mbi_hash, hicn_hash, hash_lookup_type, err_msg)
-            raise BBMyMedicareCallbackCrosswalkUpdateException(err_msg)
-
+        # fhir_id can not change for an existing user!
         if user.crosswalk.fhir_id != fhir_id:
-            err_msg = "Found user's fhir_id did not match"
-            log_get_and_update_user(auth_flow_dict, "FAIL", user, fhir_id, mbi_hash, hicn_hash, hash_lookup_type, err_msg)
-            raise BBMyMedicareCallbackCrosswalkUpdateException(err_msg)
+            mesg = "Found user's fhir_id did not match"
+            log_get_and_update_user(auth_flow_dict, "FAIL", user, fhir_id, mbi_hash, hicn_hash,
+                                    hash_lookup_type, crosswalk_updated, mesg)
+            raise BBMyMedicareCallbackCrosswalkUpdateException(mesg)
 
+        # Did the hicn change?
+        if user.crosswalk.user_hicn_hash != hicn_hash:
+            crosswalk_updated = crosswalk_updated + "H"
```

Comment:
Couple of notes here:
Do you think we could just make it an array of full name values? Just makes to easier to read. ex: `["HICN", "MBI"]`
Also for all the values, can we just go ahead and create constants? ie: 
```
HICN_UPDATED_TEXT = "HICN"

crosswalk_updated = crosswalk_updated + [HICN_UPDATED_TEXT]
```

Reply:
Another option could be creating boolean values for each:
```
hicn_updated: true
mbi_updated: false
```

(Then you wont need the 3rd option because you can infer from those two)